src/test/java/org/jabref/logic/shared/DBMSProcessorTest.java

```diff
@@ -95,6 +95,37 @@ void testInsertEntry() throws SQLException {
         assertEquals(expectedFieldMap, actualFieldMap);
     }
 
+    @Test
+    void testInsertEntryWithEmptyFields() throws SQLException {
+        BibEntry expectedEntry = new BibEntry(StandardEntryType.Article);
+
+        dbmsProcessor.insertEntry(expectedEntry);
+
+        BibEntry emptyEntry = getBibEntryExample();
+        emptyEntry.getSharedBibEntryData().setSharedID(1);
+        dbmsProcessor.insertEntry(emptyEntry); // does not insert, due to same sharedID.
+
+        Map<String, String> actualFieldMap = new HashMap<>();
+
+        try (ResultSet entryResultSet = selectFrom("ENTRY", dbmsConnection, dbmsProcessor)) {
+            assertTrue(entryResultSet.next());
+            assertEquals(1, entryResultSet.getInt("SHARED_ID"));
+            assertEquals("article", entryResultSet.getString("TYPE"));
+            assertEquals(1, entryResultSet.getInt("VERSION"));
+            assertFalse(entryResultSet.next());
+
+            try (ResultSet fieldResultSet = selectFrom("FIELD", dbmsConnection, dbmsProcessor)) {
+                while (fieldResultSet.next()) {
+                    actualFieldMap.put(fieldResultSet.getString("NAME"), fieldResultSet.getString("VALUE"));
+                }
+            }
+        }
+
+        Map<String, String> expectedFieldMap = expectedEntry.getFieldMap().entrySet().stream().collect(Collectors.toMap((entry) -> entry.getKey().getName(), Map.Entry::getValue));
```

Comment:
Isn't this always an empty list (as expected entry is empty)?

Reply:
Hm, seems like I need to adapt the queries a bit. Didn't think about this,
I should check for the shared entry id to be present.

Reply:
To be honest, I've no idea what the test is doing. You add an empty entry, then some other completely unrelated entry, and then compare against the empty entry again.